offload_vio/device_tx/plugin.cpp

```diff
@@ -0,0 +1,93 @@
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+#include "common/phonebook.hpp"
+
+#include <ecal/ecal.h>
+#include <ecal/msg/protobuf/publisher.h>
+#include <opencv2/core/mat.hpp>
+
+#include "vio_input.pb.h"
+
+using namespace ILLIXR;
+
+class offload_writer : public plugin {
+public:
+    offload_writer(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+    { 
+		// Initialize eCAL and create a protobuf publisher
+		eCAL::Initialize(0, NULL, "VIO Offloading Sensor Data Writer");
+		publisher = eCAL::protobuf::CPublisher<vio_input_proto::IMUCamVec>("vio_input");
+
+	}
+
+
+    virtual void start() override {
+        plugin::start();
+
+        sb->schedule<imu_cam_type>(id, "imu_cam", [this](switchboard::ptr<const imu_cam_type> datum, std::size_t) {
```

Comment:
Move to constructor, no need to override `start`.

Reply:
This actually resulted in an error in the server side on things when moved to the constructor. Somehow the callback was being triggered before any data was written to vio_input. This was fine on the device which is the file you commented on but I left both overriding start() just for consistency

Reply:
Say that in a comment. `// This can't go in the constructor because ...`

Reply:
Dont think so. Both of these plugins that are overriding start() are synchronously subscribed and if the plugin were a threadloop, it would have to define _p_one_iteration.

Reply:
Oh, right, sorry, forgot that `_p_thread_setup()` is only for `threadloop`s.

The integrators, for example, subscribe in their constructors. That should be the default.